Logging/FileLogHandler.cs

```diff
@@ -84,15 +83,14 @@ public void Trace(string text)
         /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
         /// </summary>
         /// <filterpriority>2</filterpriority>
-        public void Dispose()
+        public override void Dispose()
         {
-            lock (_lock)
+            base.Dispose();
+
+            if (_writer.IsValueCreated)
             {
-                if (_writer.IsValueCreated)
-                {
-                    _disposed = true;
-                    _writer.Value.Dispose();
-                }
+                _disposed = true;
```

Comment:
I have a sneaking suspicion that we've introduced a potential race by removing the synchronization on dispose. While it's incredibly unlikely (even impossible in this code base) for multiple threads to call dispose at the same time, it's arguably negligible on performance to leave it in. Locked objects that are not in contention are pretty cheap to acquire, especially when the acquire count is once (one app exit)